### PR TITLE
style: formatting improvements

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

- removed some `# fmt: off`s where they weren't needed
- changed a few string concatenations to use f-strings, primarily those that were already using f-strings and/or those that are easier to read if f-strings are used

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: disnake/message.py [hunk 1]
```diff
@@ -1179,35 +1179,28 @@ def clean_content(self) -> str:
             or remove markdown then use :func:`utils.escape_markdown` or :func:`utils.remove_markdown`
             respectively, along with this function.
         """
-        # fmt: off
         transformations = {
-            re.escape(f'<#{channel.id}>'): '#' + channel.name
-            for channel in self.channel_mentions
+            re.escape(f"<#{channel.id}>"): "#" + channel.name for channel in self.channel_mentions
```
[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/message.py]: https://github.com/DisnakeDev/disnake/pull/490/commits/efbcb44554aefa68ed56df60c07a22fcadde53c5

**Review by onerandomusername — COMMENTED** [review]
this looks good, but have to check and see if anything is missing that should be in this pull.

**Comment by onerandomusername** [comment]
One thing missing as far as I could tell, the repr for [GuildScheduledEvent](https://github.com/shiftinv/disnake/blob/8e359a580fdcdc7943d3398b666eb79bc19c457f/disnake/guild_scheduled_event.py#L213-L222) should be like most other reprs instead of an inline for loop.

**Comment by shiftinv** [comment]
> One thing missing as far as I could tell, the repr for [GuildScheduledEvent](https://github.com/shiftinv/disnake/blob/8e359a580fdcdc7943d3398b666eb79bc19c457f/disnake/guild_scheduled_event.py#L213-L222) should be like most other reprs instead of an inline for loop.

Yup, updated in 80b72abed5a01cea8c94ac0b3d061904041641e4 + cc21d2367bce7ef95010942bbf497b8e3383760e

File: disnake/ext/commands/errors.py [hunk 2]
```diff
@@ -708,7 +707,7 @@ def __init__(self, missing_roles: SnowflakeList) -> None:
         missing = [f"'{role}'" for role in missing_roles]
 
         if len(missing) > 2:
-            fmt = "{}, or {}".format(", ".join(missing[:-1]), missing[-1])
+            fmt = f"{', '.join(missing[:-1])}, or {missing[-1]}"
```
Comment (onerandomusername) [inline disnake/ext/commands/errors.py:710]: Might be worth making a helper method at the top of the file for these or is it not worth that?

File: disnake/ext/commands/help.py [hunk 3]
```diff
@@ -419,18 +419,18 @@ def get_command_signature(self, command):
...
         if len(command.aliases) > 0:
             aliases = "|".join(command.aliases)
             fmt = f"[{command.name}|{aliases}]"
             if parent_sig:
-                fmt = parent_sig + " " + fmt
+                fmt = f"{parent_sig} {fmt}"
             alias = fmt
         else:
-            alias = command.name if not parent_sig else parent_sig + " " + command.name
+            alias = command.name if not parent_sig else f"{parent_sig} {command.name}"
```
Comment (onerandomusername) [inline disnake/ext/commands/help.py:433]: this is not the same equivalent.

File: disnake/guild_scheduled_event.py [hunk 4]
```diff
@@ -211,15 +211,22 @@ def _update(self, data: GuildScheduledEventPayload) -> None:
...
+            ("channel_id", self.channel_id),
+            ("name", self.name),
+            ("description", self.description),
+            ("scheduled_start_time", self.scheduled_start_time),
+            ("scheduled_end_time", self.scheduled_end_time),
+            ("privacy_level", self.privacy_level),
+            ("status", self.status),
+            ("entity_type", self.entity_type),
+            ("entity_metadata", self.entity_metadata),
+            ("creator", self.creator),
```
Comment (onerandomusername) [inline disnake/guild_scheduled_event.py:226]: do we intend to be missing created_at? Not sure.

File: docs/extensions/attributetable.py [hunk 5]
```diff
@@ -103,9 +103,7 @@ def parse_name(self, content):
             if not modulename:
                 modulename = self.env.ref_context.get("py:module")
         if modulename is None:
-            raise RuntimeError(
-                "modulename somehow None for %s in %s." % (content, self.env.docname)
-            )
+            raise RuntimeError("modulename somehow None for {content} in {self.env.docname}.")
```
Comment (onerandomusername) [inline docs/extensions/attributetable.py:106]: missing f-string

**Review by onerandomusername — CHANGES_REQUESTED** [review]
see above comments

File: disnake/guild_scheduled_event.py (hunk 4 above)
Comment (shiftinv) [reply disnake/guild_scheduled_event.py:226]: Don't think so, I'm not seeing any other `__repr__`s that reflect the `created_at` value

File: disnake/ext/commands/errors.py (hunk 2 above)
Comment (shiftinv) [reply disnake/ext/commands/errors.py]: Sounds good.
e7d942b803b7dc853936e3cb805a1f801ea779cb

File: docs/extensions/attributetable.py (hunk 5 above)
Comment (shiftinv) [reply docs/extensions/attributetable.py]: edd7a3b0510254f6ac56466391a425351b4857dc

File: disnake/ext/commands/errors.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/ext/commands/errors.py]: Missing L842 and L846

**Review by onerandomusername — APPROVED** [review]
let's merge this

**Pull request merged**